Fails to compile when 'a lifetime parameter used

Reproduction:


```
use postgres_from_row::FromRow;

#[derive(FromRow)]
struct User<'a> {
    username: &'a str,
    email: &'a str,
}
```


Error:

```
error[E0496]: lifetime name `'a` shadows a lifetime name that is already in scope
 --> example-project\src\main.rs:4:10
  |
4 | #[derive(FromRow)]
  |          ^^^^^^^ lifetime `'a` already in scope
5 | struct User<'a> {
  |             -- first declared here
  |
  = note: this error originates in the derive macro `FromRow` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Macro expansion when using `'a` (does not compile):
```
#![feature(prelude_import)]
#[prelude_import]
use std::prelude::rust_2021::*;
#[macro_use]
extern crate std;
use postgres_from_row::FromRow;
struct User<'a> {
    username: &'a str,
    email: &'a str,
}
impl<'a> postgres_from_row::FromRow for User<'a>
where
    &'a str: for<'a> postgres_from_row::tokio_postgres::types::FromSql<'a>,
    &'a str: for<'a> postgres_from_row::tokio_postgres::types::FromSql<'a>,
{
    fn from_row(row: &postgres_from_row::tokio_postgres::Row) -> Self {
        Self {
            username: postgres_from_row::tokio_postgres::Row::get::<
                &str,
                &'a str,
            >(row, "username"),
            email: postgres_from_row::tokio_postgres::Row::get::<
                &str,
                &'a str,
            >(row, "email"),
        }
    }
    fn try_from_row(
        row: &postgres_from_row::tokio_postgres::Row,
    ) -> std::result::Result<Self, postgres_from_row::tokio_postgres::Error> {        
        Ok(Self {
            username: postgres_from_row::tokio_postgres::Row::try_get::<
                &str,
                &'a str,
            >(row, "username")?,
            email: postgres_from_row::tokio_postgres::Row::try_get::<
                &str,
                &'a str,
            >(row, "email")?,
        })
    }
}
```

Surprisingly, it compiles when you use `'b` instead of `'a` in the struct:

```
use postgres_from_row::FromRow;

#[derive(FromRow)]
struct User<'b> {
    username: &'b str,
    email: &'b str,
}
```

Macro expansion when using `'b` (compiles):
```
#![feature(prelude_import)]
#[prelude_import]
use std::prelude::rust_2021::*;
#[macro_use]
extern crate std;
use postgres_from_row::FromRow;
struct User<'b> {
    username: &'b str,
    email: &'b str,
}
impl<'b> postgres_from_row::FromRow for User<'b>
where
    &'b str: for<'a> postgres_from_row::tokio_postgres::types::FromSql<'a>,
    &'b str: for<'a> postgres_from_row::tokio_postgres::types::FromSql<'a>,
{
    fn from_row(row: &postgres_from_row::tokio_postgres::Row) -> Self {
        Self {
            username: postgres_from_row::tokio_postgres::Row::get::<
                &str,
                &'b str,
            >(row, "username"),
            email: postgres_from_row::tokio_postgres::Row::get::<
                &str,
                &'b str,
            >(row, "email"),
        }
    }
    fn try_from_row(
        row: &postgres_from_row::tokio_postgres::Row,
    ) -> std::result::Result<Self, postgres_from_row::tokio_postgres::Error> {        
        Ok(Self {
            username: postgres_from_row::tokio_postgres::Row::try_get::<
                &str,
                &'b str,
            >(row, "username")?,
            email: postgres_from_row::tokio_postgres::Row::try_get::<
                &str,
                &'b str,
            >(row, "email")?,
        })
    }
}
```

--- a/postgres-from-row-derive/src/lib.rs
+++ b/postgres-from-row-derive/src/lib.rs
@@ -170,13 +170,13 @@ impl FromRowField {
 
     /// Pushes the needed where clause predicates for this field.
     ///
-    /// By default this is `T: for<'z> postgres::types::FromSql<'z>`,
+    /// By default this is `T: for<'__from_row_lifetime> postgres::types::FromSql<'__from_row_lifetime>`,
     /// when using `flatten` it's: `T: postgres_from_row::FromRow`
     /// and when using either `from` or `try_from` attributes it additionally pushes this bound:
     /// `T: std::convert::From<R>`, where `T` is the type specified in the struct and `R` is the
     /// type specified in the `[try]_from` attribute.
     ///
-    /// Note: 'z is used here to avoid conflicts with common user-specified lifetimes like 'a
+    /// Note: '__from_row_lifetime is used here to avoid conflicts with common user-specified lifetimes like 'a
     fn add_predicates(&self, predicates: &mut Vec<TokenStream2>) -> Result<()> {
         let target_ty = &self.target_ty()?;
         let ty = &self.ty;
@@ -184,7 +184,7 @@ impl FromRowField {
         predicates.push(if self.flatten {
             quote! (#target_ty: postgres_from_row::FromRow)
         } else {
-            quote! (#target_ty: for<'z> postgres_from_row::tokio_postgres::types::FromSql<'z>)
+            quote! (#target_ty: for<'__from_row_lifetime> postgres_from_row::tokio_postgres::types::FromSql<'__from_row_lifetime>)
         });
 
         if self.from.is_some() {

--- a/postgres-from-row-derive/src/lib.rs
+++ b/postgres-from-row-derive/src/lib.rs
@@ -170,11 +170,13 @@ impl FromRowField {
 
     /// Pushes the needed where clause predicates for this field.
     ///
-    /// By default this is `T: for<'a> postgres::types::FromSql<'a>`,
+    /// By default this is `T: for<'z> postgres::types::FromSql<'z>`,
     /// when using `flatten` it's: `T: postgres_from_row::FromRow`
     /// and when using either `from` or `try_from` attributes it additionally pushes this bound:
     /// `T: std::convert::From<R>`, where `T` is the type specified in the struct and `R` is the
     /// type specified in the `[try]_from` attribute.
+    ///
+    /// Note: 'z is used here to avoid conflicts with common user-specified lifetimes like 'a
     fn add_predicates(&self, predicates: &mut Vec<TokenStream2>) -> Result<()> {
         let target_ty = &self.target_ty()?;
         let ty = &self.ty;
@@ -182,7 +184,7 @@ impl FromRowField {
         predicates.push(if self.flatten {
             quote! (#target_ty: postgres_from_row::FromRow)
         } else {
-            quote! (#target_ty: for<'a> postgres_from_row::tokio_postgres::types::FromSql<'a>)
+            quote! (#target_ty: for<'z> postgres_from_row::tokio_postgres::types::FromSql<'z>)
         });
 
         if self.from.is_some() {

